npm/private/repository_label_store.bzl

```diff
@@ -0,0 +1,115 @@
+"""In order to avoid late restarts, repository rules should pre-compute dynamic labels and
```

Comment:
TODO to upstream this to bazel-lib? Nothing JS/pnpm specific about it. Also looks nicely unit-testable

Reply:
Yea, I was thinking it would end up there. Have to learn how to unit-test repository rule code tho since this one depends on `rctx.path`

Reply:
https://github.com/aspect-build/rules_js/issues/693